### PR TITLE
csound: add Java bindings

### PR DESCRIPTION
**Pull request by nwhetsell**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This pull request adds Java bindings for Csound.

### CODE REVIEW

File: Formula/csound.rb [hunk 1]
```diff
@@ -14,7 +14,9 @@ class Csound < Formula
   depends_on "cmake" => :build
   depends_on "python" => [:build, :test]
   depends_on "python@2" => [:build, :test]
+  depends_on "swig" => :build
   depends_on "fltk"
+  depends_on :java
```
Comment (SMillerDev) [inline Formula/csound.rb]: Is this always needed? Or could this just be like python?

File: Formula/csound.rb [hunk 2]
```diff
@@ -39,6 +40,13 @@ def install
     mkdir "build" do
       system "cmake", "..", *args
       system "make", "install"
+
+      macho = MachO.open("lib_jcsound6.jnilib")
+      macho.change_install_name("@rpath/libcsnd6.6.0.dylib", "#{opt_prefix}/lib/libcsnd6.dylib")
+      macho.change_install_name("@rpath/CsoundLib64.framework/Versions/6.0/CsoundLib64", "#{opt_prefix}/Frameworks/CsoundLib64.framework/CsoundLib64")
```
Comment (SMillerDev) [inline Formula/csound.rb]: Why is this needed? What's wrong with rpath?

File: Formula/csound.rb (hunk 1 above)
Comment (nwhetsell) [reply Formula/csound.rb]: It isn’t always needed. I’ve changed this to `depends_on :java => [:build, :test]`.

File: Formula/csound.rb (hunk 2 above)
Comment (nwhetsell) [reply Formula/csound.rb]: The most honest answer I can give, unfortunately, is: I really don’t know. The Mach-O stuff isn’t necessary on my machine. With all the Mach-O stuff deleted, I can run—

```sh
brew install --build-from-source csound
brew test csound
```

—and the [Java bindings test](https://github.com/Homebrew/homebrew-core/pull/40327/files#diff-287bbbfba06a8ff0471bc82fcefc18eaR109) works fine.

The Mach-O stuff [wasn’t part](https://github.com/Homebrew/homebrew-core/pull/40327/commits/69e630a63d31e251013f1bc72daaeb31559b8392#diff-287bbbfba06a8ff0471bc82fcefc18eaR42) of this pull request originally, but the Java bindings test failed on Jenkins. (It seems like these test results should be available as [build #42678](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/42678/), but I get a 404 error. Builds between [#38238](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/38238/) and [#43089](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/43089/) appear to be missing.) If you want, I can remove the Mach-O stuff and see what the error was on Jenkins.

Comment (SMillerDev) [reply Formula/csound.rb]: Please do.

And old builds get removed so will respond with 404 after a while.

Comment (nwhetsell) [reply Formula/csound.rb]: After removing the Mach-O stuff, the test fails with:

```
==> java -Djava.library.path='/usr/local/opt/csound/libexec' test
Picked up _JAVA_OPTIONS:  -Duser.home=/Users/brew/Library/Caches/Homebrew/java_cache
_jcsound6 native code library failed to load.
java.lang.UnsatisfiedLinkError: /usr/local/Cellar/csound/6.12.2_3/libexec/lib_jcsound6.jnilib: dlopen(/usr/local/Cellar/csound/6.12.2_3/libexec/lib_jcsound6.jnilib, 1): Library not loaded: @rpath/CsoundLib64.framework/Versions/6.0/CsoundLib64
  Referenced from: /usr/local/Cellar/csound/6.12.2_3/libexec/lib_jcsound6.jnilib
  Reason: image not found
```

Comment (nwhetsell) [reply Formula/csound.rb]: Is there anything I can do to help get this merged?

The fact that I can’t reproduce the Java link error locally is weird. I suppose I could just remove the test of Java bindings; I don’t know of another formula that tests these.

**Comment by fxcoudert** [comment]
> The fact that I can’t reproduce the Java link error locally is weird. I suppose I could just remove the test of Java bindings

We can't accept them if they're not tested, otherwise they will bitrot

**Comment by nwhetsell** [comment]
Should I just close this pull request then? I haven’t been able to reproduce what’s breaking on Jenkins locally, and I don’t know enough about what may be different on Jenkins to even make a guess about what the issue is.

**Comment by fxcoudert** [comment]
Yes, I guess. I'm sorry we couldn't be of more help…

**Comment by nwhetsell** [comment]
No worries, thanks for the review.
